### ISSUE TITLE
Support placeholders in InfinispanRemoteAutoConfiguration

### ISSUE THREAD

**Issue by jpechane** [infinispan/infinispan-spring-boot#9]
If a ISPN client properties is loaded it is loaded via standard java.util.Properties. Thus the application needs to hard-code for example initial server list. It would be useful if a Spring mechanism for loading properties that supports placeholders would be used. This would allow the developer to refer to env vars/system properties and thus be provided during deployment time. Typical use case are env vars provided by Kuberentes that refer to a running ISPN server service.

**Comment by tristantarrant** [comment]
https://issues.jboss.org/browse/ISPN-7419
Since it is very easy to implement, I should have something ready soon.

**Comment by slaskawi** [comment]
Actually this can be done purely in Spring. 

Spring supports injecting ([please have a look here for more information](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html#boot-features-external-config)). Then you need to take this property and manually create a `InfinispanRemoteConfigurer` as shows [here](https://github.com/slaskawi/presentations/blob/DevConf_final/2016_devconf/antifroud-checker/src/main/java/org/infinispan/microservices/transactions/configuration/TransactionConfiguration.java#L21-L29).

Here's a copied example:

```
@Value("${spring.infinispan.server}")
private String infinispanAddress;

@Bean
public InfinispanRemoteConfigurer infinispanRemoteConfigurer() {
   return () -> new ConfigurationBuilder()
         .addServers(infinispanAddress)
         .build();
}
```

**Comment by jpechane** [comment]
I know and I use it - but still it is kind of a code duplication. And parameter tuning means code recompilation (if I need to add a new one) instead of just a change in deployment.

**Comment by slaskawi** [comment]
Granted. In that case [ISPN-7419](https://issues.jboss.org/browse/ISPN-7419) could be a solution for your problem or alternatively we could have another look at https://github.com/infinispan/infinispan-spring-boot/issues/8. 

The only problem with https://github.com/infinispan/infinispan-spring-boot/issues/8 is that we have [lots of those properties](https://github.com/infinispan/infinispan/blob/master/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/ConfigurationProperties.java) and this add some maintenance work on our side to keep them in sync.

**Comment by slaskawi** [comment]
@jpechane This should already be available in `9.0.0.CR1`. We recently released `9.0.0.CR2`. Could you please check if it works and if it solves your needs?
